packages/graphql/src/schema-model/relationship/Relationship.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Neo4jGraphQLSchemaValidationError } from "../../classes";
+import { upperFirst } from "../../utils/upper-first";
+import type { Attribute } from "../attribute/Attribute";
+import type { ConcreteEntity } from "../entity/ConcreteEntity";
+import type { Entity } from "../entity/Entity";
+
+export type RelationshipDirection = "IN" | "OUT";
+
+export class Relationship {
+    public readonly name: string;
+    public readonly type: string;
+    public readonly attributes: Map<string, Attribute> = new Map();
+    public readonly source: ConcreteEntity; // Origin field of relationship
+    public readonly target: Entity;
+    public readonly direction: RelationshipDirection;
+
+    /**Note: Required for now to infer the types without ResolveTree */
+    public get connectionFieldTypename(): string {
```

Comment:
This is interesting, but I'm not 100% sure about it, do we want to start adding a dependency of the augmentedSchema generated types in the schema model? Not as a blocker but as a conversation point for the whole IR design!

Reply:
My idea is to have a set of classes that given a schema model generates the augmented schema and some stubs that can be used to collect these infos, it's just a rough idea after I saw this use case, we can discuss this in some  IR DDD sessions!